### ISSUE TITLE
Thunderbird appears to leak memory when using Conversations

### ISSUE THREAD

**Issue by Slater91** [thunderbird-conversations/thunderbird-conversations#1757]
* **Thunderbird version:** 91.3.1
* **Thunderbird Conversation version:** 4.0.5
* **Operating System:** Linux (KDE neon, based on Ubuntu 20.04)

**What did you do?**
I used Thunderbird to read and send emails throughout the day on 7 email addresses (plus two configured, but not used).

**What did you expect to happen?**
I expected memory usage to remain more-or-less constant: when Thunderbird opens, it uses around 300 MB of RAM.

**What actually happened?**
Memory usage kept on increasing, to the point it went well over 1 GB at the end of the day. Disabling Conversations significantly reduces the memory impact, to the point memory usage was around 400 MB after three days of consecutive usage and without program restarts. This is quite a significant difference and is always reproducible on my systems (both my desktop and my laptop display the same behaviour).
I understand that Conversations is an extremely complex add-on and that increased memory usage is to be expected, but only up to a certain point. The constant increment in usage sounds like a memory leak to me, but I can only take a guess from the outside.
I also understand that my description lacks any kind of technical details that would allow for a deeper understanding of the issue, so please do let me know if there is anything I can do to better identify the issue behind this behaviour.
Thanks!

**Comment by Standard8** [comment]
Thank you for the report.

I did a bit of digging and discovered that deselecting messages or changing folders would trigger this. Turns out that whilst we were cleaning up properly when different messages were selected, we weren't when the message display goes blank (e.g. change folder etc) and hence the leak.

I've pushed a fix. I want to try and get a few other things fixed as well, so it might be a week or so before this gets released.
